pkg/api/dashboard.go

```diff
@@ -122,7 +122,7 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 		meta.FolderUrl = query.Result.GetUrl()
 	}
 
-	provisioningData, err := dashboards.NewProvisioningService().GetProvisionedDashboardDataByDashboardID(dash.Id)
+	provisioningData, err := dashboards.NewService(hs.DataService).GetProvisionedDashboardDataByDashboardID(dash.Id)
```

Comment:
Can you please keep the `NewProvisioningService` factory method that returns a `DashboardProvisioningService`? This function can call `NewService`. This way we're only exposing the dashboard provisioning interface in the provisioning package and has a smaller foot-print and doesn't risk that provisioning calling the non-provisioning methods on the dashboard service. 

The same reasoning goes for someone having a dependency on the dashboard service shouldn't be able to call provisioning methods. This was the historic reasoning for keeping DashboardService and DashboardProvisioningService separate.

Reply:
@marefr OK, will do.